lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResultImpl.cs

```diff
@@ -43,20 +45,57 @@ internal class HttpClientHelper : IDriverHttpEndpoint
         private const string AppJson = "application/json";
 
         private string _driverUrl;
+        private string _appId;
 
         private readonly HttpClient _client;
+        private readonly IREEFClient _reefClient;
 
-        internal HttpClientHelper()
+        internal JobSubmissionResultImpl(IREEFClient reefClient, string filePath, bool yarn)
         {
+            _reefClient = reefClient;
             _client = new HttpClient
             {
                 Timeout = TimeSpan.FromSeconds(SecondsForHttpClientTimeout),
             };
             _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(AppJson));
+
+            if (yarn)
```

Comment:
This is a very dangerous design, as it enumerates all possible runtimes in a way that will be hard to find if we ever want to support e.g. Mesos in the C# code. A better way would be to add a new interface for this that the runtime in question needs to implement & bind. That way, new runtime authors have clear(er) guidance on what they need to do.